Integrate nrfx 2.0.0

anangl opened the pull request:
This PR updates the hal_nordic module revision and introduces several adjustments needed to switch to nrfx 2.0.0.

---
See also https://github.com/BabbleSim/ext_NRF52_hw_models/pull/1.

kl-cruz committed:
drivers: wdt_nrfx: Add multi-instance support

Update the driver to support multiple watchdog instances
and add the corresponding Kconfig options.

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>
Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
soc: nordic_nrf: Update validation of base adresses with new entries

Add checks for:
- NRF_RNG
- NRF_IPC
- NRF_WDT0
- NRF_WDT1

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
modules: hal_nordic: Update nrfx to version 2.0.0

Update the hal_nordic module revision to switch to nrfx 2.0.0.
Add Kconfig options that enable to use the newly introduced nrfx
drivers in Zephyr.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
drivers: nrfx: Update implementations after switching to nrfx 2.0.0

Update calls to nrfx HAL functions to reflect API changes introduced in
nrfx 2.0.0. All these functions are now called with the first parameter
pointing to the structure of registers of the relevant peripheral.
Also a few functions got renamed:
- nrf_gpiote_int_is_enabled to nrf_gpiote_int_enable_check
- nrf_gpiote_event_is_set to nrf_gpiote_event_check
- nrf_rng_event_get to nrf_rng_event_check
- nrf_rng_int_get to nrf_rng_int_enable_check
- nrf_rtc_event_pending to nrf_rtc_event_check
- nrf_rtc_int_is_enabled to nrf_rtc_int_enable_check
- nrf_timer_cc_read to nrf_timer_cc_get
- nrf_timer_cc_write to nrf_timer_cc_set

Default configuration values were removed from nrfx_config files,
so the drivers pwm_nrfx and spi_nrfx_spis no longer can use those.

Function nrfx_pwm_init() now takes one more parameter - context pointer
that is passed to the event handler, not used in the pwm_nrfx driver.

HALs for UART and UARTE now allow configuration of the parity type
and the number of stop bits, for SoCs that provide the corresponding
registers.

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>
Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
bluetooth: nordic: Update nrfx HAL calls with pointers to HW instances

Update calls to nrfx HAL functions to reflect API changes introduced in
nrfx 2.0.0. All these functions are now called with the first parameter
pointing to the structure of registers of the relevant peripheral.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
nordic: Update nrfx HAL function calls after switching to nrfx 2.0.0

Calls to nrfx HAL functions in various nRF platform related source
files are complemented with pointers to relevant peripherals.

Additionally, TIMER HAL functions that got renamed in nrfx 2.0.0 are
updated in the qemu_cortex_m0 board supporting code.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
samples/boards/nrf91/nrfx: Align implementation with nrfx 2.0.0

Complement GPIOTE HAL function calls with the NRF_GPIOTE parameter.
Update nrfx_gpiote_init() function call with the newly introduced
parameter.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ioannisg reviewed (approved):
I suppose the block refers to the PR for BabbleSim?

anangl commented:
> I suppose the block refers to the PR for BabbleSim?

Exactly.

tbursztyka commented on drivers/watchdog/Kconfig.nrfx:
```diff
@@ -10,3 +10,16 @@ config WDT_NRFX
 	help
 	  Enable support for nrfx WDT driver for nRF MCU series.
+
+config NRFX_WDT0
```
Making Kconfig aware of instances is something we should not do anymore. It should be the sole role of DTS now. 

I know it is still present in many places, and that is required to be fixed. But let's not introduce new ones.

Could look at it?

anangl replied:
I agree, but this is consistent with what we do for all other nrfx based drivers. This is obviously to be corrected and I have it on my to-do list. But I think it will be better to do the clean up for all these drivers together, instead of introducing the new approach in this PR (this one is for getting nrfx 2.0.0 integrated).

tbursztyka replied:
ok

anangl committed:
nrf52_bsim: Update the NRF52 HW models revision

Update the models revision used for building so that nrf52_bsim
can be used with nrfx 2.0.0.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ioannisg commented:
@kl-cruz can you take a look here?

aescolar reviewed (approved):
Ok for simulation

ioannisg commented on boards/posix/nrf52_bsim/hw_models_version:
```diff
@@ -1 +1 @@
-v1.8
+test_2019_11_08_CIc
```
@aescolar @anangl can this actually be merged as is? Just double-checking

carlescufi replied:
Yes, this is fine, we checked with @aescolar offline

ioannisg replied:
oh great

aescolar replied:
I will send a trivial fix right after this is merged updating it to v2.0 (they both point to the same sha)
